concent_api/verifier/utils.py

```diff
@@ -354,3 +357,28 @@ def download_archives_from_storage(file_transfer_token, subtask_id, package_path
                 ErrorCode.VERIFIER_FILE_DOWNLOAD_FAILED,
                 subtask_id=subtask_id,
             )
+
+
+def parse_result_files_with_frames(frames, result_files_list, output_format):
+    parsed_files_to_compare = {}
+    for frame_number in frames:
+        for result_file_name in result_files_list:
+            if re.search(f'_{frame_number:>04}.{output_format.lower()}', result_file_name) is not None:
+                if not parsed_files_to_compare.get(output_format):
+                    parsed_files_to_compare.update({frame_number: [generate_verifier_storage_file_path(result_file_name)]})
+                    result_files_list.remove(result_file_name)
+    return parsed_files_to_compare
+
+
+def render_images_by_frames(parsed_files_to_compare, frames, output_format, scene_file, subtask_id):
+    blender_output_file_name_list = []
+    for frame_number in frames:
+        render_image(frame_number, output_format, scene_file, subtask_id)
+        blender_output_file_name_list.append(generate_full_blender_output_file_name(scene_file, frame_number, output_format.lower()))
+        parsed_files_to_compare[frame_number].append(generate_full_blender_output_file_name(scene_file, frame_number, output_format.lower()))
```

Comment:
please extract `generate_full_blender_output_file_name(scene_file, frame_number, output_format.lower())` to a variable and reuse

Reply:
Done